### PR TITLE
[dev] Add `dev.portal/diff->`, drop-in `->` that sends diffs to Portal

### PR DESCRIPTION
**Pull request by bshepherdson**
### Description

Adds `dev.portal/diff->`, a drop-in replacement for `->` that `tap>`s
a diff at each stage. (With the metadata for Portal to default to diff viewer)

I found myself wanting this several times while debugging the refs overhaul.

### How to verify

Try it out at dev time!

### Checklist

- [ ] Tests have been added/updated to cover changes in this PR
    - Skipping; this is a dev helper

### CODE REVIEW

**Comment by bshepherdson** [comment]
* **#50470** <a href="https://app.graphite.dev/github/pr/metabase/metabase/50470?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/metabase/metabase/50470?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `master`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: dev/src/dev/portal.clj [hunk 1]
```diff
@@ -68,3 +71,20 @@
...
     (send-log event)))
+
+(defmacro diff->
+  "Drop-in replacement for `->` that sends diffs to Portal at each stage."
+  [x & forms]
+  #_{:clj-kondo/ignore [:discouraged-var]}
+  (loop [x x, forms forms]
+    (if forms
+      (let [form (first forms)
+            threaded `(-> ((fn [before#]
```
Comment (crisptrutski) [inline dev/src/dev/portal.clj:82]: You don't need the outer threading macro here, it's just taking the one form.

**Review by crisptrutski — APPROVED** [review]
Nifty!

File: dev/src/dev/portal.clj [hunk 2]
```diff
@@ -1,6 +1,8 @@
 (ns dev.portal
   (:require [portal.api :as p]))
 
+(set! *warn-on-reflection* true)
```
Comment (crisptrutski) [inline dev/src/dev/portal.clj:4]: Why was this required? I don't see any interop

Comment (bshepherdson) [reply dev/src/dev/portal.clj:4]: It's [not in the diff](https://github.com/metabase/metabase/pull/50470/files/53802612c8f01e94a303d7521e97419b617e760b#diff-022365b6dcb53c61bbe8524b40fed905a99f833b37b836c22cdc0457726613ebR72), I just angered the linter gods.

File: dev/src/dev/portal.clj (hunk 1 above)
Comment (bshepherdson) [reply dev/src/dev/portal.clj:82]: I do, if I want this to really be a drop-in `->`, because `->` also handles (1) propagating metadata and (2) wrapping bare values in lists.

Comment (crisptrutski) [reply dev/src/dev/portal.clj:82]: Isn't the inner `->` on line 83 taking care of that? I'm talking about the one on line 82, which will just short circuit to identity.

Comment (metamben) [reply dev/src/dev/portal.clj:82]: I agree with @crisptrutski, the outer `->` seem to be a no-op.

**Review by metamben — APPROVED** [review]
